packages/components/nodes/outputparsers/StructuredOutputParserAdvanced/StructuredOutputParserAdvanced.ts

```diff
@@ -0,0 +1,80 @@
+import { getBaseClasses, INode, INodeData, INodeParams } from '../../../src'
+import { BaseOutputParser } from 'langchain/schema/output_parser'
+import { StructuredOutputParser as LangchainStructuredOutputParser } from 'langchain/output_parsers'
+import { CATEGORY } from '../OutputParserHelpers'
+import { z } from 'zod'
+
+class AdvancedStructuredOutputParser implements INode {
+    label: string
+    name: string
+    version: number
+    description: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    inputs: INodeParams[]
+    credential: INodeParams
+
+    constructor() {
+        this.label = 'Advanced Structured Output Parser'
+        this.name = 'advancedStructuredOutputParser'
+        this.version = 1.0
+        this.type = 'AdvancedStructuredOutputParser'
+        this.description = 'Parse the output of an LLM call into a given structure by providing a Zod schema.'
+        this.icon = 'structure.svg'
+        this.category = CATEGORY
+        this.baseClasses = [this.type, ...getBaseClasses(BaseOutputParser)]
+        this.inputs = [
+            {
+                label: 'Autofix',
+                name: 'autofixParser',
+                type: 'boolean',
+                optional: true,
+                description: 'In the event that the first call fails, will make another call to the model to fix any errors.'
+            },
+            {
+                label: 'Example JSON',
+                name: 'exampleJson',
+                type: 'string',
+                description: 'Zod schema for the output of the model',
+                rows: 10,
+                default: `z.object({
+    title: z.string(), // Title of the movie as a string
+    yearOfRelease: z.number().int(), // Release year as an integer number,
+    genres: z.enum([
+        "Action", "Comedy", "Drama", "Fantasy", "Horror",
+        "Mystery", "Romance", "Science Fiction", "Thriller", "Documentary"
+    ]).array().max(2), // Array of genres, max of 2 from the defined enum
+    shortDescription: z.string().max(500) // Short description, max 150 characters
+})`,
+                additionalParams: true
```

Comment:
Think we can probably don't display as additional parameter. Would like to display it on the node itself, since user must enter the code specifying the zod schema

Reply:
Fair enough! Updated it